common/models/acl.js

```diff
@@ -376,6 +376,15 @@ module.exports = function(ACL) {
   };
 
   /**
+   * Test if ACL permission is not DENY
+   * @returns {Boolean} is permission type different than 'DENY' ?
+   */
+
+  ACL.prototype.isAllowed = function() {
+    return this.permission !== loopback.ACL.DENY;
```

Comment:
> I'm wondering of a case where the interim result of ACL scoring would be something like {role1: {score: 900, permission: 'ALLOW'}, role2: {score: 800, permission: 'DENY'}, role3: {score: 700, permission: 'ALLOW'}}, and what should be the content of authorizedRoles in such a case.

I think there should be a test case to show what's the expected result in this case.

I would expect that such request will be allowed, and my expectation as a LoopBack user would be that `role2` is not included in `authorizedRoles`.

If I understand/remember your code correctly, if the permission was `ACL.DEFAULT` and `model.settings.defaultPermission` was set to `DENY`, then `authorizedRoles` will include `role2`. Again, there should be a test exercising this scenario and verifying that an expected outcome is achieved.